src/mon_markings.c

```diff
@@ -0,0 +1,383 @@
+{
+    u16 i;
+    sMenu->cursorPos = 0;
+    sMenu->markings = markings;
+    for (i = 0; i < 4; i++)
+        sMenu->markingsArray[i] = (sMenu->markings >> i) & 1;
+    DestroyMenuCursor();
+    sub_80F761C(x, y, sMenu->baseTileTag, sMenu->basePaletteTag);
+}
+
+void sub_80F7470(void)
+{
+    u16 i;
+
+    for (i = 0; i < 3; i++)
+    {
+        FreeSpriteTilesByTag(sMenu->baseTileTag + i);
+    }
+
+    FreeSpritePaletteByTag(sMenu->basePaletteTag);
+    FreeSpritePaletteByTag(sMenu->basePaletteTag + 1);
+
+    for (i = 0; i < 2; i++)
+    {
+        if (!sMenu->menuWindowSprites[i])
+            return;
+        DestroySprite(sMenu->menuWindowSprites[i]);
+    }
+
+    for (i = 0; i < 4; i++)
+    {
+        if (!sMenu->menuMarkingSprites[i])
+            return;
+        DestroySprite(sMenu->menuMarkingSprites[i]);
+    }
+
+    DestroyMenuCursor();
+
+    if (sMenu->menuTextSprite)
+        DestroySprite(sMenu->menuTextSprite);
+}
+
+bool8 sub_80F7500(void)
+{
+    u16 i;
+    struct SpriteSheet sheets[3] =
+    {
+        { sMenu->menuWindowSpriteTiles, 0x1000, sMenu->baseTileTag },
+        { gPokenavConditionMenuMisc_Gfx, 0x520, sMenu->baseTileTag + 1 },
+        { NULL, 0, 0 }
+    };
+
+    if (sMenu->spriteSheetLoadRequired)
+    {
+        LoadTilesForSpriteSheets(sheets);
+        sMenu->spriteSheetLoadRequired = FALSE;
+    }
+
+    if (gMain.newKeys & DPAD_UP)
+    {
+        s8 pos;
+        PlaySE(SE_SELECT);
+        pos = --sMenu->cursorPos;
+        if (pos < 0)
+            sMenu->cursorPos = 5;
+        return TRUE;
+    }
+
+    if (gMain.newKeys & DPAD_DOWN)
+    {
+        s8 pos;
+        PlaySE(SE_SELECT);
+        pos = ++sMenu->cursorPos;
+        if (pos > 5)
+            sMenu->cursorPos = 0;
+        return TRUE;
+    }
+
+    if (gMain.newKeys & A_BUTTON)
+    {
+        PlaySE(SE_SELECT);
+
+        switch (sMenu->cursorPos)
+        {
+        case 4:
+            sMenu->markings = 0;
+            for (i = 0; i < 4; i++)
+                sMenu->markings |= sMenu->markingsArray[i] << i;
+            return FALSE;
+        case 5:
+            return FALSE;
+        }
+
+        sMenu->markingsArray[sMenu->cursorPos] = sMenu->markingsArray[sMenu->cursorPos] ? FALSE : TRUE;
+        return TRUE;
+    }
+
+    if (gMain.newKeys & B_BUTTON)
+    {
+        PlaySE(SE_SELECT);
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+void sub_80F761C(s16 x, s16 y, u16 baseTileTag, u16 basePaletteTag)
+{
+    u16 i;
+    u8 spriteId;
+
+    struct SpriteSheet sheets[] =
+    {
+        { sMenu->menuWindowSpriteTiles, 0x1000, baseTileTag },
+        { gPokenavConditionMenuMisc_Gfx, 0x520, baseTileTag + 1 },
+        { NULL, 0 }
+    };
+
+    struct SpritePalette palettes[] =
+    {
+        { sMenu->framePalette, basePaletteTag },
+        { gUnknown_08E966B8, basePaletteTag + 1},
+        { NULL, 0 }
+    };
+
+    struct SpriteTemplate sprTemplate;
+    struct SpriteTemplate *sprTemplatePtr = &sprTemplate;
```

Comment:
This matches without the `sprTemplatePtr` variable if you initialize `sprTemplate` like this:
```
struct SpriteTemplate sprTemplate =
    {
        baseTileTag,
        basePaletteTag,
        &gOamData_83E5214,
        gSpriteAnimTable_83E52AC,
        NULL,
        gDummySpriteAffineAnimTable,
        nullsub_65,
    };
```